### PR TITLE
sycl: update documentation

### PR DESCRIPTION
**Pull request by sgeor255**
This PR updates the documentation about SYCL primitives.

### CODE REVIEW

File: src/gpu/generic/sycl/README.md [hunk 1]
```diff
@@ -47,50 +92,77 @@ The implementation supports both forward and backward directions.
...
 ## LRN
 
 The implementation supports both forward and backward directions.
 
 * Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`, `NC`
+* Supported data types: `f32`, `bf16`, `f16`
+
+## Matmul
+
+* Supported formats: plain formats
```
Comment (densamoilov) [inline src/gpu/generic/sycl/README.md:116]: Should we mention something about leveraging the cublaslt format when format tag is any?

File: src/gpu/generic/sycl/README.md [hunk 2]
```diff
@@ -33,11 +33,56 @@ one, the environment variable can be set to `cuda:*`.
...
+* Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`;
+* Supported data types: `f32`, `bf16`, `f16`, `s32`, `s8`, `u8`
+* Limitations
+    * Some very large problem sizes currently return `unimplemented` due to an
+      issue with long execution times
+
+## Concat
+
+A generic Concat implementation which uses the Reorder primitive is used. Refer
+to [src/gpu/generic/ref_concat.hpp](../ref_concat.hpp) for more information.
```
Comment (densamoilov) [inline src/gpu/generic/sycl/README.md:73]: I think it's an implementation detail that may not be very useful for the users. Each primitive can have multiple implementations so I believe we just need to document what the primitive supports and not a particular implementation. When it makes sense we can always document peculiarities that can be related to a certain piece of hardware (e.g. A100 matmul, etc).

File: src/gpu/generic/sycl/README.md (hunk 1 above)
Comment (densamoilov) [reply src/gpu/generic/sycl/README.md:116]: Nevermind, I confused the documentations for NVIDIA and generic vendors :)

File: src/gpu/generic/sycl/README.md (hunk 2 above)
Comment (sgeor255) [reply src/gpu/generic/sycl/README.md]: Good point, added details.

**Review by densamoilov — APPROVED** [review]

**Review by mgouicem — APPROVED** [review]

File: src/gpu/generic/sycl/README.md [hunk 3]
```diff
@@ -33,11 +33,56 @@ one, the environment variable can be set to `cuda:*`.
...
+## Convolution
+
+The implementation supports forward, backward data and backward weights
+directions.
+
+* Supported input/output formats: plain formats
+* Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`;
+* Supported data types: `f32`, `bf16`, `f16`, `s32`, `s8`, `u8`
+* Limitations
+    * Some very large problem sizes currently return `unimplemented` due to an
```
Comment (ranukund) [inline src/gpu/generic/sycl/README.md:67]:
```suggestion
    * Few substantial problem sizes currently return `unimplemented` due to an
```

File: src/gpu/generic/sycl/README.md [hunk 4]
```diff
@@ -33,11 +33,56 @@ one, the environment variable can be set to `cuda:*`.
...
+* Supported formats: plain formats
+* Supported data types: `f32`, `bf16`, `f16`, `s8`, `s32`
+
+## Deconvolution
+
+The implementation supports forward and backward data and backward weights
+directions.
+
+* Supported input/output formats: plain formats
+* Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`;
```
Comment (ranukund) [inline src/gpu/generic/sycl/README.md:81]:
```suggestion
* Supported weights formats: `goiw`, `goihw`, `goidhw`, `oiw`, `oihw`, `oidhw`
```

File: src/gpu/generic/sycl/README.md [hunk 5]
```diff
@@ -33,11 +33,56 @@ one, the environment variable can be set to `cuda:*`.
...
+* Supported data types: `f32`, `bf16`, `f16`, `s32`, `s8`, `u8`
+* Limitations
+    * Some very large problem sizes currently return `unimplemented` due to an
```
Comment (ranukund) [inline src/gpu/generic/sycl/README.md:84]:
```suggestion
    * Few substantial problem sizes currently return `unimplemented` due to an
```

File: src/gpu/generic/sycl/README.md [hunk 6]
```diff
@@ -47,50 +92,77 @@ The implementation supports both forward and backward directions.
...
+* Limitations
+    * Runtime dims is not supported
+    * PReLU post-op is not supported
 
 ## Pooling
 
 The implementation supports both forward and backward directions.
 
 * Supported formats: `NCDHW`, `NDHWC`, `NCHW`, `NHWC`, `NCW`, `NWC`
+* Supported data types for forward direction: f32, bf16, f16, s8, u8
```
Comment (ranukund) [inline src/gpu/generic/sycl/README.md:127]:
```suggestion
* Supported data types for forward direction: `f32`, `bf16`, `f16`, `s8`, `u8`
```

File: src/gpu/generic/sycl/README.md [hunk 7]
```diff
@@ -47,50 +92,77 @@ The implementation supports both forward and backward directions.
...
+* Supported data types for backward direction: f32, bf16, f16
```
Comment (ranukund) [inline src/gpu/generic/sycl/README.md:128]:
```suggestion
* Supported data types for backward direction: `f32`, `bf16`, `f16`
```

**Review by ranukund — COMMENTED** [review]
Minor changes suggested. Please incorporate as you see fit.

**Review by ranukund — APPROVED** [review]

File: src/gpu/generic/sycl/README.md [hunk 8]
```diff
@@ -33,11 +33,56 @@ one, the environment variable can be set to `cuda:*`.
 
 # Supported Primitives
 
+General limitations:
+
+* Currently blocked formats are not supported by any implementation unless
+    explicitly listed
```
[the remark this answers is not in the document]
Comment (sgeor255) [reply src/gpu/generic/sycl/README.md]: Fixed, thanks.

**Comment by sgeor255** [comment]
> Minor changes suggested. Please incorporate as you see fit.

Thanks, @ranukund , I have made most of the suggested changes.

**Pull request merged**